src/App.tsx

```diff
@@ -15,36 +15,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>([...goodsFromServer]); 
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [activeSort, setActiveSort] = useState<SortType>(SortType.Default);
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort(SortType.Default);
+  };
+
+  const sortAlphabetically = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort((a, b) =>
+        a.localeCompare(b),
+      );
+
+      return isReversed ? sortedGoods.reverse() : sortedGoods;
+    });
+    setActiveSort(SortType.Alphabetical);
+  };
+
+  const sortByLength = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort(
+        (a, b) => a.length - b.length,
+      );
+
+      return isReversed ? sortedGoods.reverse() : sortedGoods;
+    });
+    setActiveSort(SortType.Length);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevReversed => !prevReversed);
+  };
```

Comment:
It isn't easy to understand what is happening here. Why in sort method you reverse and so on?

Actually, you don't need to store goods in a state at all. Store just in a variable as it's a dynamic value depending on the sort type and isReversed. Create a function that prepares goods and use it:
```typescript
const preparedGoods = prepareGoods(goods, { sortType, isReversed });
```